autogpt_platform/backend/backend/blocks/example.py

```diff
@@ -0,0 +1,160 @@
+        )
+        error: str = SchemaField(description="The error from the example block")
+
+    def __init__(self):
+        super().__init__(
+            # The unique identifier for the block, this value will be persisted in the DB.
+            # It should be unique and constant across the application run.
+            # Use the UUID format for the ID.
+            id="380694d5-3b2e-4130-bced-b43752b70de9",
+            # The description of the block, explaining what the block does.
+            description="The example block",
+            # The list of contributors who contributed to the block.
+            # Each contributor is represented by a ContributorDetails object.
+            contributors=[ContributorDetails(name="Craig Swift")],
+            # The set of categories that the block belongs to.
+            # Each category is an instance of BlockCategory Enum.
+            categories={BlockCategory.BASIC},
+            # The schema, defined as a Pydantic model, for the input data.
+            input_schema=self.ExampleBlockInput,
+            # The schema, defined as a Pydantic model, for the output data.
+            output_schema=self.ExampleBlockOutput,
+            # The list or single sample input data for the block, for testing.
+            # This is an instance of the Input schema with sample values.
+            test_input=self.ExampleBlockInput(
+                name="Craig", greeting="Hello", is_funny=True
+            ).model_dump(),
+            # The list or single expected output if the test_input is run.
+            # Each output is a tuple of (output_name, output_data).
+            test_output=[
+                ("response", {"message": "Hello, world!"}),
+            ],
+            # Function names on the block implementation to mock on test run.
+            # Each mock is a dictionary with function names as keys and mock implementations as values.
+            test_mock={
+                "my_function_that_can_be_mocked": lambda *args, **kwargs: "Hello, world!"
+            },
+            # The credentials required for testing the block.
+            # This is an instance of APIKeyCredentials with sample values.
+            test_credentials=TEST_CREDENTIALS,
+            # The type of the block, which is an instance of BlockType Enum.
+            block_type=BlockType.STANDARD,
+            # The webhook configuration for the block, if any.
+            # This can be an instance of BlockWebhookConfig or BlockManualWebhookConfig.
+            webhook_config=None,
+        )
+
+    def my_function_that_can_be_mocked(self, input: str) -> str:
+        logger.info("my_function_that_can_be_mocked called with input: %s", input)
+        return "Hello, world!"
+
+    def run(self, input_data: ExampleBlockInput, **kwargs) -> BlockOutput:
+        try:
+            message = self.my_function_that_can_be_mocked(input_data.greeting)
+            yield "response", {"message": message}
+        except Exception as e:
```

Comment:
This should be handled already I believe by the wrapper on the block, unless you want to handle different types of errors differently in which case you should re-raise cc @majdyz